src/Foundatio/Storage/FolderFileStorage.cs

```diff
@@ -88,29 +88,22 @@ public class FolderFileStorage : IFileStorage {
             path = path.NormalizePath();
 
             string file = Path.Combine(Folder, path);
+            string directory = Path.GetDirectoryName(file);
 
             try {
-                using (var fileStream = CreateFileStream(file)) {
+                using (var fileStream = File.Create(file)) {
```

Comment:
I think using the existing method, is the best (we could move it to a private non local method). This could have recursion issues.